sound/soc/sof/intel/hda-ipc.c

```diff
@@ -260,6 +263,13 @@ irqreturn_t hda_dsp_ipc_irq_handler(int irq, void *context)
 		snd_sof_dsp_update_bits_unlocked(sdev, HDA_DSP_BAR,
 						 HDA_DSP_REG_ADSPIC,
 						 HDA_DSP_ADSPIC_IPC, 0);
+		sdev->irq_event |= SOF_IRQ_IPC;
```

Comment:
the use of a field that's set in the handler is problematic. We know we can have new interrupts before the thread is scheduled (that's e.g. how SoundWire handled the commands before, with the wait_for_completion in the thread and the complete() bin the handler

So it's possible that while you have an SoundWire thread scheduled an IPC interrupt quicks in, and you would reset sdev->irq_event.

I don't think this can work...